src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -29,12 +29,15 @@ namespace ray {
 namespace gcs {
 
 GcsServer::GcsServer(const ray::gcs::GcsServerConfig &config,
-                     boost::asio::io_service &main_service)
+                     std::vector<boost::asio::io_service *> io_services)
     : config_(config),
-      main_service_(main_service),
+      main_service_(*io_services[0]),
+      node_manager_io_service_(*io_services[1]),
       rpc_server_(config.grpc_server_name, config.grpc_server_port,
                   config.grpc_server_thread_num),
-      client_call_manager_(main_service) {}
+      client_call_manager_(*io_services[0]) {
+  RAY_CHECK(io_services.size() >= 2);
```

Comment:
I think there's no way that this can be bigger than 2 now right? Let's just have a check == 2

Reply:
ok